src/library/FOSSBilling/Validate.php

```diff
@@ -43,6 +50,64 @@ public function isSldValid(string $sld): bool
         }
     }
 
+    /**
+     * Validates if a given TLD is valid, comparing against the official TLD list by the IANA.
+     * In the event that fetching the valid list doesn't work, some very simple validation is performed instead.
+     */
+    public function isTldValid(string $tld): bool
+    {
+        $tld = ltrim($tld, '.');
+        $tld = idn_to_ascii($tld);
+        $tld = strtoupper($tld);
+
+        $validTlds = $this->di['cache']->get('validTlds', function (ItemInterface $item) {
+            $item->expiresAfter(86400);
+
+            $client = HttpClient::create(['bindto' => BIND_TO]);
+            $response = $client->request('GET', 'https://data.iana.org/TLD/tlds-alpha-by-domain.txt');
```

Comment:
.com.br is a valid extension how ever data.iana.org does only lists top level domains

https://raw.githubusercontent.com/publicsuffix/list/master/public_suffix_list.dat

is a more complete list

Reply:
Good catch, I hadn't considered that.
Switched it over, works:
![image](https://github.com/FOSSBilling/FOSSBilling/assets/17304943/9abb7a49-9379-49cb-9a39-e0eed1bd95ac)